tsdb/importer/importer.go

```diff
@@ -0,0 +1,602 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importer
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/gob"
+	"io"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/textparse"
+	"github.com/prometheus/prometheus/pkg/value"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/prometheus/prometheus/tsdb/fileutil"
+)
+
+// Implementing the error interface to create a
+// constant, which cannot be overridden.
+// https://dave.cheney.net/2016/04/07/constant-errors
+type Error string
+
+func (e Error) Error() string {
+	return string(e)
+}
+
+const (
+	// This error is thrown when the current entry does not have a timestamp associated.
+	NoTimestampError = Error("expected timestamp with metric")
+	// This error is thrown when the sample being parsed currently has a corresponding block
+	// in the database, but has no metadata collected for it, curiously enough.
+	NoBlockMetaFoundError = Error("no metadata found for current samples' block")
```

Comment:
isn't " ' " notation reserved to people? Charles' hat ?

Reply:
It's the possessive generally, there's no restriction on people or proper nouns.